### PR TITLE
adding private and making tabs dynamic

### PR DESCRIPTION
**Pull request by fazanki**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
   @Input() public showCommentSummary: boolean;
   @Output() public readonly showCommentSummaryDialog = new EventEmitter();
 
-  tabs = [{label: 'comments'}, {label: 'filter'}, {label: 'search'}];
-  tabSelected: string;
+  tabs: {isFiltered?: boolean; label: string}[] = [];
+  tabSelected = 'comments';
   isFiltered: boolean;
   navigationList: Annotation[];
+  $ubscriptions: Subscription;
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts:23]: I assume this is supposed to be subscriptions$

File: projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts [hunk 2]
```diff
@@ -1,38 +1,41 @@
...
 
-    this.store.pipe(select(fromAnnoSelector.getFilteredAnnotations)).subscribe(filteredAnno => {
+    this.$ubscriptions = combineLatest([tagFilter$, filteredAnnotation$]).subscribe(([formData, filteredAnno]) => {
       this.navigationList = filteredAnno;
+      this.tabs = this.navigationList.length > 0 ?
+        [{label: 'comments'}, {label: 'filter'}, {label: 'search'}] : [{label: 'comments'}];
+      this.isFiltered = !formData.length;
+      this.tabs = [...this.tabs].map((tab) => {
+        return !this.isFiltered && tab.label === 'filter' ? {...tab, isFiltered: true} : {...tab, isFiltered: false};
+      });
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts:38]: why not do the mapping directly with the first assignment?

File: projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts (hunk 1 above)
Comment (fazanki) [reply projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts:23]: oh dear

Comment (fazanki) [reply projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts:23]: no
$subscription 
observable$

i thought i misspell it and been looking like 1/2 minute where is the typo

File: projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts (hunk 2 above)
Comment (fazanki) [reply projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts:38]: its better to decouple the reference as i am assigning the this.tabs to this.tabs

**Pull request merged**